fastlane/fastlane/Fastfile

```diff
@@ -6,28 +6,65 @@ end
 
 desc "Increment the version number of this gem"
 lane :bump do |options|
+  ensure_git_branch(branch: "master")
+  ensure_git_status_clean
+
   tool_name = options[:tool] || detect_tool
   path = "./lib/#{tool_name}/version.rb"
 
-  latest_release = current_version(tool: tool_name)
-  `open https://github.com/#{github_url(tool: tool_name)}/compare/#{latest_release}...master`
+  # Verify everything is in a consistent state
+  latest_version = current_version(tool: tool_name)
+  local_version = version_get_podspec(path: path)
+  UI.user_error!("Version on RubyGems doesn't match local repo: #{latest_version} != #{local_version}") if latest_version != local_version
 
-  local_version = version_get_podspec(path: path) # here, to not show the lane change
+  changelog_text = show_changelog
 
-  puts "Live Version: #{latest_release}"
-  puts "Local Version: #{local_version}"
+  bump_type ||= 'minor' if prompt(text: "New feature, method or API?", boolean: true)
+  bump_type ||= 'patch'
 
-  bump_type ||= 'minor' if prompt(text: "New Feature?", boolean: true)
-  bump_type ||= 'patch' if bump_type.nil? && prompt(text: "Bugfix/Improvements?", boolean: true)
+  slug = "fastlane/fastlane"
+  new_version = version_bump_podspec(path: path, bump_type: bump_type)
+  sh "git checkout -b '#{tool_name}-version-bump-#{new_version}'"
+  commit_message = "[#{tool_name}] Version bump to #{new_version}"
+  git_commit(path: path,
+             message: commit_message)
+  push_to_git_remote
 
-  if bump_type
-    version_bump_podspec(path: path, bump_type: bump_type)
-    git_commit(path: path,
-               message: "Version Bump by fastlane")
-    push_to_git_remote
-  else
-    puts "Doing nothing..."
+  github_api_token = ENV["FL_GITHUB_RELEASE_API_TOKEN"]
+  UI.user_error!("Please provide a GitHub API token using `FL_GITHUB_RELEASE_API_TOKEN`") if github_api_token.to_s.length == 0
```

Comment:
Commits and pushes are already happening before this, so the check should be at the beginning of the lane